tests/testthat/test-visualizations.R

```diff
@@ -0,0 +1,92 @@
+# Tests for visualization functions
+# library(SingleR); library(testthat); source("setup.R"); source("test-visualization.R")
+
+test.exp <- assay(test,1)
+training.exp <- assay(training,1)
+
+test_that("we can produce expression comparison scatterplots with plotCellVsReference", {
+  expect_error(plotCellVsReference(test,1,training,1), NULL) # looks for logcounts, but the dataset does not have logcounts
+  expect_s3_class(P <- plotCellVsReference(test, 1, training, 1, 1, 1), "ggplot")
+  expect_s3_class(plotCellVsReference(test.exp, 1, training.exp, 1), "ggplot")
+  expect_s3_class(plotCellVsReference(test, 1, training.exp, 1,1,NULL), "ggplot")
+  expect_s3_class(plotCellVsReference(test.exp, 1, training, 1,NULL,1), "ggplot")
+  expect_false(isTRUE(all.equal(P,
+                                P2 <- plotCellVsReference(test, 2, training, 1, 1, 1))))
+  expect_false(isTRUE(all.equal(P, 
+                                P3 <- plotCellVsReference(test, 1, training, 2, 1, 1))))
+  expect_false(isTRUE(all.equal(P2, P3)))
+})
+
+example("SingleR")
```

Comment:
I ran `example(SingleR)` to get a full SingleR output for plotScoreHeatmap testing.
I had tried using `source("test-SingleR.R")`, but it threw an error, and didn't store an output.

Reply:
_testthat_ will automatically execute `setup.R` before any `test-*.R` scripts, so you can use the stuff in there rather than running `example(SingleR)`. So in your `test-visualiations.R`, throw in a `SingleR()` call using the SCEs constructed in `setup.R`. This keeps example code separate from test code, which makes life easier for further development (e.g., your tests won't break because you renamed variables in the examples).

Reply:
Fixed.  I only use the setup data now.